extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -141,4 +152,38 @@ public void logCleanup(BuildProducer<LogCleanupFilterBuildItem> logCleanupFilter
         logCleanupFilter.produce(new LogCleanupFilterBuildItem("io.smallrye.metrics.MetricsRegistryImpl",
                 "Register metric ["));
     }
+
+    @BuildStep
+    @Record(STATIC_INIT)
+    void registerMetricsFromAnnotatedMethods(SmallRyeMetricsRecorder metrics,
+            BeanContainerBuildItem beanContainerBuildItem,
+            BeanArchiveIndexBuildItem beanArchiveIndex) {
+        JandexBeanInfoAdapter beanInfoAdapter = new JandexBeanInfoAdapter(beanArchiveIndex.getIndex());
+        JandexMemberInfoAdapter memberInfoAdapter = new JandexMemberInfoAdapter(beanArchiveIndex.getIndex());
+        for (ClassInfo clazz : beanArchiveIndex.getIndex().getKnownClasses()) {
```

Comment:
I really don't think it's a good idea to scan the whole index.

Shouldn't we interrogate the index for the annotations instead?

I'm really worried this won't scale very well when having a large application.

Reply:
I agree this may be somewhat slow for larger apps. To remedy that, I tried adding some filters to leave out classes which are unlikely to contain annotations (a few lines below - it might be a good idea to add some more filtering, as the comment suggests, I wasn't yet sure what more to add), then I scan the remaining classes for metric-related annotations. 

Switching to iterating over annotations could be a good idea, but I suppose that if a large application has a lot of classes, it could have a lot of annotations as well, so I'm not sure if it would help in all cases. So yeah you guys can experiment with that

Reply:
The big difference is that you will focus on the ones having the proper annotations.

Your applications is large but not all your classes will have the relevant annotations. That's why we have an index in the first place :).

Reply:
Good point. Actually I think that at first I was trying to do it this way, but I didn't after all, because of some issue I had, but I don't remember what it was.

Reply:
So what's the plan here?

I know it may not be ideal, but it's happening during build vs startup right?

Can we consider merging and adding a fix later?

Reply:
Or we could fix it right away.

I don't think it's overly complicated and I really don't want this in in its current form, especially since I'm not sure it will get fixed before the next release. Sure it's build time but we don't want build time to skyrocket (think dev mode for instance).

Reply:
If Jan is gone for two weeks, how do we progress this without pushing to his branch to update the PR?

Reply:
@machi1990 @gwenneg would one of you would be interested in driving this puppy home?

It's an interesting one I think but I don't know how much you already have on your respective plates :).

Reply:
@kenfinnigan well, it's just a matter of getting his branch and creating a new branch and potentially a new PR.

Reply:
Ok, wasn't sure if there was a shorter way

Reply:
Hey @kenfinnigan @gsmet I have not entirely followed the discussion but I can take a look over the weekend.

Reply:
@kenfinnigan we can just push to his PR but for that, I think you need to be a committer of the project.

Reply:
@machi1990 so the issue here is that currently it scans the whole index and then collect the appropriate annotations (see `beanArchiveIndex.getIndex().getKnownClasses()`).

What we need to do is to look for the appropriate annotations in the index (the usual pattern we applied everywhere) and act from there.

Happy to answer any questions anytime :).

Reply:
@gsmet thanks for the heads up. I'll take it from there.